Clarify purpose of encryption and disposal of disks

stilez opened the pull request:
2 bullets in this section are unclear/contradictory. For example, it states in one place that the primary purpose is secure wiping at end of use, in another that the primary purpose is against physical disk theft, and nowhere states anything about temporary (illicit) removal + viewing while powered down/"data at rest". Edited 2 paragraphs to make this consistent and clear.

Also while about it, be clearer about disposal too, as it's much the same thing, and not well explained

wblock commented on userguide/storage.rst:
```diff
@@ -217,13 +217,12 @@ Click the *+* next to the volume name to access
-   the system.
-
+.. note:: The encryption facility used by %brand% is designed to protect
+   against physical theft and removal of disks, and to prevent data recovery 
```
This sentence is kind of contradictory and confusing.  Encryption is only used to prevent data recovery, so leaving the original "theft and removal" in place is not useful.

stilez replied:
reworded - done

wblock commented on userguide/storage.rst:
```diff
@@ -217,13 +217,12 @@ Click the *+* next to the volume name to access
-
+.. note:: The encryption facility used by %brand% is designed to protect
+   against physical theft and removal of disks, and to prevent data recovery 
+   from reused or failed disks. It is not designed to protect against
```
Nonspecific "It" here should be "This type of encryption".

stilez replied:
reworded - done

wblock commented on userguide/storage.rst:
```diff
@@ -217,13 +217,12 @@ Click the *+* next to the volume name to access
+   against physical theft and removal of disks, and to prevent data recovery 
+   from reused or failed disks. It is not designed to protect against
+   unauthorized software access. Ensure that only authorized users have
+   access to the administrative GUI and that proper permissions are set
```
s/administrative GUI/|web-ui|/

stilez replied:
done

wblock commented on userguide/storage.rst:
```diff
@@ -240,15 +239,21 @@ intended use:
-  it.
+* The type of encryption used by %brand% is primarily used to protect 
+  data against the risks of disks (or the entire system) being physically
+  stolen, or removed to view their contents when the system is turned off.
```
There is a weird break with "a system being stolen or removed".  The comma does not help, making the sentence longer but not clearer.

stilez replied:
rephrased - with added serial commas!

wblock commented on userguide/storage.rst:
```diff
@@ -240,15 +239,21 @@ intended use:
+  data against the risks of disks (or the entire system) being physically
+  stolen, or removed to view their contents when the system is turned off.
+  Because data cannot be read without the key, it also permits disks
+  containing sensitive data to be safely removed, repurposed or disposed
```
Use the serial comma, always.

stilez replied:
Mwah - "But I can't help it if US English gets commas wrong".....    ;-)
*chuckle*
https://en.wikipedia.org/wiki/Serial_comma

wblock commented on userguide/storage.rst:
```diff
@@ -240,15 +239,21 @@ intended use:
+  stolen, or removed to view their contents when the system is turned off.
+  Because data cannot be read without the key, it also permits disks
+  containing sensitive data to be safely removed, repurposed or disposed
+  of, without secure wiping or physical destruction of the magnetic platters.
```
Replace unpleasant "disposed of" with "discarded" or some other simpler synonym.

wblock commented on userguide/storage.rst:
```diff
@@ -240,15 +239,21 @@ intended use:
+  containing sensitive data to be safely removed, repurposed or disposed
+  of, without secure wiping or physical destruction of the magnetic platters.
+
+* When disposing of disks that still contain encrypted sensitive data,
```
As above, s/disposing of/discarding/

stilez replied:
done

wblock commented on userguide/storage.rst:
```diff
@@ -240,15 +239,21 @@ intended use:
+  of, without secure wiping or physical destruction of the magnetic platters.
+
+* When disposing of disks that still contain encrypted sensitive data,
+  the encryption key must also be destroyed or securely deleted, or if
```
Break long sentence after "deleted".

wblock commented on userguide/storage.rst:
```diff
@@ -240,15 +239,21 @@ intended use:
+
+* When disposing of disks that still contain encrypted sensitive data,
+  the encryption key must also be destroyed or securely deleted, or if
+  kept, then it must be held independently and securely. As long as 
```
s/then//

What does "independently" mean here?  "Kept separate"?

stilez replied:
reworded + done

wblock commented on userguide/storage.rst:
```diff
@@ -240,15 +239,21 @@ intended use:
+* When disposing of disks that still contain encrypted sensitive data,
+  the encryption key must also be destroyed or securely deleted, or if
+  kept, then it must be held independently and securely. As long as 
+  a copy of the key is intact, part or all of the stored data will
```
s/will remain/remains/  (use present tense)

wblock commented on userguide/storage.rst:
```diff
@@ -240,15 +239,21 @@ intended use:
+  the encryption key must also be destroyed or securely deleted, or if
+  kept, then it must be held independently and securely. As long as 
+  a copy of the key is intact, part or all of the stored data will
+  remain vulnerable to being decrypted.
```
s/being decrypted/decryption/

stilez replied:
done

stilez commented:
@wblock - All requested changes done. Please review.

stilez commented on userguide/storage.rst:
```diff
@@ -218,11 +218,12 @@ Encryption
-   unauthorized software access. Ensure that only authorized users have
-   access to the administrative GUI and that proper permissions are set
-   on shares if sensitive data is stored on the system. 
+   against data theft from powered-down systems and locked encrypted pools,
```
reworded - any better this time?

stilez commented on userguide/storage.rst:
```diff
@@ -218,11 +218,12 @@ Encryption
+   and to prevent data recovery from reused or failed disks. This type of data
+   encryption is not designed to protect against unauthorized software access
+   when the pool is already unlocked. Ensure that only authorized users have
+   access to the web UI and that proper permissions are set on shares if sensitive
```
Are those "|" a literal part of the markup?

stilez commented on userguide/storage.rst:
```diff
@@ -239,18 +240,18 @@ intended use:
-* The type of encryption used by %brand% is primarily used to protect 
-  data against the risks of disks (or the entire system) being physically
-  stolen, or removed to view their contents when the system is turned off.
+* The type of encryption used by %brand% is primarily intended to protect
```
I've considerably refactored this part as a result. I've mergewd it withj thew notew at the top of the section, to removce duplication and to be clearer "up front", and also it seemed a sensible place to rewference GELI as the underlying scheme, rather than talk about encryption and only later say what is used.  Should be more straightforward.

stilez commented on userguide/storage.rst:
```diff
@@ -239,18 +240,18 @@ intended use:
+  powered down, when the pool is locked, or in the event of physical theft.
   Because data cannot be read without the key, it also permits disks
-  containing sensitive data to be safely removed, repurposed or disposed
+  containing sensitive data to be safely removed, repurposed, or disposed
```
done

stilez commented on userguide/storage.rst:
```diff
@@ -239,18 +240,18 @@ intended use:
-  remain vulnerable to being decrypted.
+* When discarding disks that still contain encrypted sensitive data,
+  the encryption key must also be destroyed or securely deleted.  If
+  kept, then it should be stored securely and separately from the system.
```
done

stilez commented on userguide/storage.rst:
```diff
@@ -239,18 +240,18 @@ intended use:
+* When discarding disks that still contain encrypted sensitive data,
+  the encryption key must also be destroyed or securely deleted.  If
+  kept, then it should be stored securely and separately from the system.
+  As long as a copy of the key is intact, part or all of the stored data remains
```
done. Edited "can be" to "will be" (because it's correct, they will be)

stilez commented:
Made 4 other changes not listed by @wblock :+1: 

- **Line 234** - "is right" seems informal and vague. Changed to "is an appropriate measure"

- **Line 240 - added "and pools"** - entire pools are encrypted. So if an encrypted pool uses a partition on a drive, and an unencrypted pool uses a 2nd partition on the same drive, encryption would be pereforemed "per pool", hence not necessarily the entirety of all disks, it would seem.

- **Line 249**, seems the key point here is "someone can get hold of the key, who also has access to the drives". It's not simply where the key is stored. A trusted person might have access to the backup server or office or filing cabinet for example. I've tried to reword to fix this, without adding undue length.

- **Line 265-267 - encrypted pool elements** - we've hammered this out elsewhere, and concluded that L2ARC is encrypted, if the underlying pool is encrypted. Fixing here too.

stilez commented:
Fixed a merge conflict, by rewording the paragraph to flow better and be more direct. 

I also moved the paragraph concerned to before the bullet list, because it's a basic explanation of encrypted disks, not a supplementary detail needed to understand whether encryption is useful and how to use it correctly.

Last, I improved the paragraph on encrypting an existing pool. If the user has other disks, or an existing encrypted pool, then the instructions to backup and then restore are pointless - they can just copy it over. Also it should be more clear that any unencrypted versions "left over" should be wiped, not just disks reformatted (which is what "destroying the original pool" does).

Mrt134 commented on userguide/storage.rst:
```diff
@@ -217,42 +217,50 @@ Click the *+* next to the volume name to access
+  the same person who gains access to the disks, the data will be 
+  vulnerable to decryption.
+   
+* Protect the key with a strong passphrase and store all backups of
```
```suggestion
* Protect the key with a strong passphrase and store all key backups
```

Mrt134 commented on userguide/storage.rst:
```diff
@@ -217,42 +217,50 @@ Click the *+* next to the volume name to access
+  vulnerable to decryption.
+   
+* Protect the key with a strong passphrase and store all backups of
+  the key securely. If the encryption key is lost, the data on the disks is
```
```suggestion
  securely. If the encryption key is lost, the data on the disks is
```

Mrt134 commented on userguide/storage.rst:
```diff
@@ -261,23 +269,20 @@ intended use:
-  unencrypted volume. Instead, the data must be backed up, the
-  existing pool destroyed, a new encrypted volume created, and the
-  backup restored to the new volume.
+* At present, there is no one-step way to encrypt an unencrypted volume. Instead, the data must
```
```suggestion
* At present, there is no one-step way to encrypt an unencrypted volume. The data must
```

Mrt134 commented on userguide/storage.rst:
```diff
@@ -261,23 +269,20 @@ intended use:
 
-* Hybrid pools are not supported. Added vdevs must match the existing
+* Hybrid volumes are not supported. Added vdevs must match the existing
   encryption scheme. :ref:`Volume Manager` automatically encrypts a new
```
```suggestion
  encryption scheme. :ref:`Volume Manager` automatically encrypts new
```

Mrt134 commented on userguide/storage.rst:
```diff
@@ -261,23 +269,20 @@ intended use:
+* Hybrid volumes are not supported. Added vdevs must match the existing
   encryption scheme. :ref:`Volume Manager` automatically encrypts a new
-  vdev being added to an existing encrypted pool.
+  vdev being added to an existing encrypted volume.
```
```suggestion
  vdevs added to an existing encrypted volume.
```

dlavigne commented:
Angulargui commit: https://github.com/freenas/freenas-docs/commit/da6e520a0d364675d66a2ee8f314228f536e1283#diff-8516e547e07e4197eeaeba1236246c9e
